File: 1.two-sum/solution.erl
-spec two_sum(Nums :: [integer()], Target :: integer()) -> [integer()].
two_sum(Nums, Target) ->
  A = lists:zip(Nums, lists:seq(0, length(Nums) - 1)),
  lists:foldl(fun({Num, Idx}, AccIn) ->
    case AccIn of
      undefined ->
        case lists:search(fun({Num2, Idx2}) ->
          Idx /= Idx2 andalso Target == Num2 + Num
        end, A) of
          {value, {Num1, Idx1}} ->
            [Idx, Idx1];
          false ->
            undefined
        end;
      Other ->
        Other
    end
  end, undefined, A).

File: 1.two-sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
  const s = new Map<number, number>();
  for (let i = 0; i < nums.length; i++) {
    const index2 = s.get(target - nums[i]);
    if (index2 !== undefined) {
      return [index2, i];
    }
    s.set(nums[i], i);
  }
  return [];
};